crypto/keys/private.go

```diff
@@ -23,8 +24,16 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
+
+	"github.com/google/trillian"
 )
 
+// Provider handles acquisition of signers for trees.
+type Provider interface {
+	// Signer returns a signer for the given tree.
+	Signer(context.Context, *trillian.Tree) (crypto.Signer, error)
```

Comment:
I was cautious about putting "New" at the beginning of this. It's arguably more of a "GetSigner" method, in that it goes and acquires a private key or surrogate to use as a signer. Go style doesn't really believe in "Get" prefixes though.